api/common/v1alpha1/headermatch.go

```diff
@@ -0,0 +1,16 @@
+package v1alpha1
```

Comment:
@michaelbeaumont what do you think about sharing this type with MeshTrafficRoute policy?

Reply:
Yeah makes sense I think.